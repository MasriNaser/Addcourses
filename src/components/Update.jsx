//@flow
import React, { Component } from 'react';
import { Button, Form, Input } from 'semantic-ui-react';
import PropTypes from 'prop-types';

export default class Update extends Component {
  render() {
    const { defaultValue, onSubmit, onChange } = this.props;
    return (
      <Form onSubmit={onSubmit}>
        <div className='ui focus input'>
          <Input
            type='text'
            placeholder={defaultValue}
            onChange={onChange}
            autoFocus
            required
          />
        </div>
        <Button>update Course</Button>
      </Form>
    );
  }
}
Update.prototypes = {
  defaultValue: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};