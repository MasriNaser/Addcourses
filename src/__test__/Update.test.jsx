import React from 'react';
import { shallow } from 'enzyme';
import Update from '../components/update';
import toJson from 'enzyme-to-json';

const onDelete = jest.fn().mockName('onDelete');
const onEdit = jest.fn().mockName('onEdit');
const wrapper = shallow(<Update updateForm={onDelete} name='' onEdit={onEdit} />);
describe('updateForm component', () => {
  it('should take snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should has Form', () => {
    expect(wrapper.find('Form').exists()).toBe(true);
  });
  it('should has input', () => {
    expect(wrapper.find('input').exists()).toBe(true);
  });
  it('should has Button', () => {
    expect(wrapper.find('Button').exists()).toBe(true);
  });
  
});
