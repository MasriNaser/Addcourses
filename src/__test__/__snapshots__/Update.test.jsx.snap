// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`updateForm component should take snapshot 1`] = `
<Form
  as="form"
>
  <div
    className="ui focus input"
  >
    <Input
      autoFocus={true}
      required={true}
      type="text"
    />
  </div>
  <Button
    as="button"
  >
    update Course
  </Button>
</Form>
`;
